on-premise-docker-registry/README.md

```diff
@@ -0,0 +1,191 @@
+- [htpasswd](https://httpd.apache.org/docs/2.4/programs/htpasswd.html) installed
+
+## Setup
+
+The commands have been verified on OSX. However, it should be possible to adapt them for a windows laptop.
+
+### Environment variables
+
+- Export the following environment variables:
+
+```shell
+export KUBECONFIG=<path-to-kubeconfig>
+export NAMESPACE={kyma-namespace-used-for-this-sample}
+export CLUSTER_DOMAIN={kyma-cluster-domain}
+export REG_USER_NAME={docker-registry-user}
+export REG_USER_PASSWD={docker-registry-password}
+export EMAIL={your-email}
+```
+
+### On-premise Docker registry
+
+In this sample, we set up a simple Docker registry running on your machine. This can be replaced with a productive alternate deployed in your corporate network. The only requirement is that it should be reachable using Cloud Connector.
+
+- Generate the self-signed certificate for enabling HTTPS for the Docker registry. Be sure to specify the `CN` as `myregistry.kyma`
+
+```shell
+make generate-self-signed-cert
+```
+
+- Add the newly created certificate to your trust storage
+
+```shell
+make trust-self-signed-cert
+```
+
+- Generate the `htpasswd`. This is used for accessing the on-premise Docker registry.
+
+```shell
+make generate-htpasswd
+```
+
+- Start the Docker registry server
+
+```shell
+make start-docker-registry
+```
+
+- Add DNS entry to the `/etc/hosts` file
+
+```shell
+127.0.0.1 myregistry.kyma
+```
+
+- Configure Cloud Connector to access the on-premise Docker registry.
+  ![cc-config](assets/cc-config.png)
+
+## nginx as reverse proxy
+
+We use Nginx as a reverse proxy to forward the HTTP requests for pulling Docker images from the on-premise Docker registry using Connectivity Proxy and Cloud Connector.
+
+It is exposed as the `NodePort` service. This exposes the Service on the Kubernetes worker node on a port. Kubelet is a component running on each Kubernetes worker node. Among other tasks, it is responsible for pulling the Docker images.
+
+When creating a deployment, we specify the Docker registry as `localhost:{NodePort}`. This is the address of the nginx reverse proxy. The nginx reverse proxy then forwards the call to the on-premise Docker registry using Connectivity Proxy and Cloud Connector.
+
+This sample shows a simple configuration with nginx as a reverse proxy. You can use any other reverse proxy implementation based on your on-premise Docker registry behavior and APIs.
+
+- Create namespace and enable istio sidecar injection if not done earlier.
+
+```shell
+kubectl create namespace ${NAMESPACE}
+kubectl label namespace ${NAMESPACE} istio-injection=enabled
+```
+
+- Deploy the nginx as a reverse proxy. The following components will be deployed:
+  - [ConfigMap](./k8s/configmap.yaml) for nginx configuration
+  - [Deployment and Service](./k8s/deployment.yaml)
+  - [PeerAuthentication](./k8s/peer-authentication.yaml) set to `PERMISSIVE` to allow for communication between kubelet and the nginx reverse proxy.
+
+```shell
```

Comment:
add 3 blank spaces at the beginning of each line of the code block for proper indentation below the list item.